src/core/client/admin/routes/Configure/sections/Auth/SSOKeyRotation/RotationDropdown.tsx

```diff
@@ -0,0 +1,72 @@
+import { Localized } from "@fluent/react/compat";
+import React, { FunctionComponent } from "react";
+
+import {
+  Button,
+  ClickOutside,
+  Dropdown,
+  DropdownButton,
+  Icon,
+  Popover,
+} from "coral-ui/components/v2";
+
+import RotateOption, { RotateOptions } from "./RotationOption";
+
+import styles from "./RotationDropdown.css";
+
+interface Props {
+  onRotateKey: (rotation: string) => void;
+  disabled?: boolean;
+}
+
+const RotationDropDown: FunctionComponent<Props> = ({
+  onRotateKey,
+  disabled,
+}) => {
+  return (
+    <Localized
+      id="configure-auth-sso-rotate-dropdown-description"
+      attrs={{ description: true }}
+    >
+      <Popover
+        id="sso-key-rotate"
+        placement="bottom-start"
+        description="A dropdown to rotate the SSO key"
+        body={({ toggleVisibility }) => (
+          <ClickOutside onClickOutside={toggleVisibility}>
+            <Dropdown>
+              {Object.keys(RotateOptions).map((opt: string) => (
+                <DropdownButton
+                  key={opt}
+                  onClick={() => {
+                    onRotateKey(opt);
+                    toggleVisibility();
```

Comment:
Should be wrapped in a `useCallback`.

Reply:
It can't, it's using a `toggleVisibility` func that is coming from the Popover's body lambda.